src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -27,27 +30,18 @@ public override bool TryMapFromProjectedDocumentRange(RazorCodeDocument codeDocu
                 throw new ArgumentNullException(nameof(projectedRange));
             }
 
-            originalRange = default;
-
-            var csharpSourceText = SourceText.From(codeDocument.GetCSharpDocument().GeneratedCode);
-            var range = projectedRange;
-            var startIndex = range.Start.GetAbsoluteIndex(csharpSourceText);
-            if (!TryMapFromProjectedDocumentPosition(codeDocument, startIndex, out var hostDocumentStart, out var _))
+            if (mappingBehavior == MappingBehavior.Strict)
```

Comment:
Nit: Switch instead of if-else-if.

Reply:
Eh not a fan for bool statements like this.